src/components/check-your-phone/check-your-phone-controller.ts

```diff
@@ -145,134 +128,119 @@ export function checkYourPhoneGet(req: Request, res: Response): void {
-      email,
-      otp: code,
-    };
-    let isPhoneNumberUpdated = false;
-    let changePhoneNumberWithMfaApiErrorMessage: string | undefined = undefined;
-
-    if (supportChangeMfa(req.cookies)) {
-      const mfaClient = await createMfaClient(req, res);
-      const changePhoneNumberResult = await changePhoneNumberwithMfaApi(
-        mfaClient,
-        code,
-        intent,
-        newPhoneNumber,
-        res.locals.trace,
-        req.session.mfaMethods,
-        req.t
-      );
-      isPhoneNumberUpdated = changePhoneNumberResult.success;
-      if (changePhoneNumberResult.success === false) {
-        changePhoneNumberWithMfaApiErrorMessage =
-          changePhoneNumberResult.errorMessage;
-      }
-    } else {
-      isPhoneNumberUpdated = await service.updatePhoneNumber(
-        { ...updateInput, updatedValue: newPhoneNumber },
-        await getRequestConfigFromExpress(req, res)
-      );
-    }
-
-    if (isPhoneNumberUpdated) {
-      updateSessionUser(req, newPhoneNumber, getUserJourney(intent));
-      return res.redirect(getRedirectUrl(intent));
-    }
+  if (code.length > 6) {
+    const error = formatValidationError(
+      "code",
+      req.t("pages.checkYourPhone.code.validationError.maxLength")
+    );
+    return renderBadRequest(
+      res,
+      req,
+      TEMPLATE_NAME,
+      error,
+      getRenderOptions(req, intent)
+    );
+  }
 
+  if (code.length < 6) {
     const error = formatValidationError(
       "code",
-      changePhoneNumberWithMfaApiErrorMessage ??
-        req.t("pages.checkYourPhone.code.validationError.invalidCode")
+      req.t("pages.checkYourPhone.code.validationError.minLength")
+    );
+    return renderBadRequest(
+      res,
+      req,
+      TEMPLATE_NAME,
+      error,
+      getRenderOptions(req, intent)
     );
+  }
 
-    renderBadRequest(
+  if (!containsNumbersOnly(code)) {
+    const error = formatValidationError(
+      "code",
+      req.t("pages.checkYourPhone.code.validationError.invalidFormat")
+    );
+    return renderBadRequest(
       res,
       req,
       TEMPLATE_NAME,
       error,
       getRenderOptions(req, intent)
     );
-  };
+  }
+
+  const { newPhoneNumber } = req.session.user;
+
+  logger.info(
+    { trace: res?.locals?.trace },
+    `Check your phone POST controller newPhoneNumber: ${
+      req.session.user.newPhoneNumber?.replace(
+        /^(.{2})(.*)/,
+        (_, first2, rest) => first2 + rest.replace(/./g, "*")
+      ) ?? JSON.stringify(req.session.user.newPhoneNumber)
+    }`
+  );
+
+  let isPhoneNumberUpdated = false;
+  let changePhoneNumberWithMfaApiErrorMessage: string | undefined = undefined;
+
+  const mfaClient = await createMfaClient(req, res);
+  const changePhoneNumberResult = await changePhoneNumberwithMfaApi(
+    mfaClient,
+    code,
+    intent,
+    newPhoneNumber,
+    res.locals.trace,
+    req.session.mfaMethods,
+    req.t
+  );
+  isPhoneNumberUpdated = changePhoneNumberResult.success;
```

Comment:
I think we can get rid of this `isPhoneNumberUpdated` variable completely now and move the success state up to the top to get the early return in first.